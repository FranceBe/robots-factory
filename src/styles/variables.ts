/* eslint-disable sort-keys */

// Colors to use matching the graphic chart
export const palette: { [key: string]: string } = {
  blue_primary: '#279AF1',
  blue_secondary: '#1b6ca9',
  pink_primary: '#DB93B0',
  pink_secondary: '#99677b',
  green_primary: '#45B69C',
  red_primary: '#F45B69',
  grey_primary: '#6A6C71',
  grey_secondary: '#b3b3b3',
  grey_tertiary: '#D4D5D6',
  grey_quaternary: '#E5E5E5',
  white: '#FFF',
}

// Spaces to use to respect an 8px grid
export const spaces: { [key: string]: string } = {
  x_small: '4px',
  small: '8px',
  regular: '16px',
  medium: '24px',
  large: '32px',
  x_large: '40px',
}

// Font sizes to use in project
export const fontSizes: { [key: string]: string } = {
  large: '1.5rem',
  medium: '1.25rem',
  regular: '1rem',
  small: '0.75rem',
}
