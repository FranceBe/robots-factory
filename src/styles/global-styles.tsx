import { createGlobalStyle } from 'styled-components'
import { fontSizes, palette, spaces } from 'styles/variables'

import RobotoRegular from '../../static/assets/fonts/Roboto-Regular.ttf'

export const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: 'Roboto';
    src: url(${RobotoRegular}) format('ttf');
    font-style: normal;
    font-weight: 400;
  }
  body {
    font-family: 'Roboto', sans-serif;
    font-size: ${fontSizes.regular};
    margin: ${spaces.medium};
    background-color: ${palette.grey_quaternary};
    color: ${palette.grey_primary};
  }
  
`
