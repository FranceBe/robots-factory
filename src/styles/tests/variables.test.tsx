import { fontSizes, palette, spaces } from 'styles/variables'

describe('variables', () => {
  describe('palette', () => {
    it('should match snapshot', () => {
      expect(palette).toMatchSnapshot()
    })
  })
  describe('fontSizes', () => {
    it('should match snapshot', () => {
      expect(fontSizes).toMatchSnapshot()
    })
  })
  describe('spaces', () => {
    it('should match snapshot', () => {
      expect(spaces).toMatchSnapshot()
    })
  })
})
