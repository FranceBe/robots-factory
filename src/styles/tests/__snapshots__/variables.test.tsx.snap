// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variables fontSizes should match snapshot 1`] = `
Object {
  "large": "1.5rem",
  "medium": "1.25rem",
  "regular": "1rem",
  "small": "0.75rem",
}
`;

exports[`variables palette should match snapshot 1`] = `
Object {
  "blue_primary": "#279AF1",
  "blue_secondary": "#1b6ca9",
  "green_primary": "#45B69C",
  "grey_primary": "#6A6C71",
  "grey_quaternary": "#E5E5E5",
  "grey_secondary": "#b3b3b3",
  "grey_tertiary": "#D4D5D6",
  "pink_primary": "#DB93B0",
  "pink_secondary": "#99677b",
  "red_primary": "#F45B69",
  "white": "#FFF",
}
`;

exports[`variables spaces should match snapshot 1`] = `
Object {
  "large": "32px",
  "medium": "24px",
  "regular": "16px",
  "small": "8px",
  "x_large": "40px",
  "x_small": "4px",
}
`;
